model/common/src/icon4py/model/common/grid/geometry_program.py

```diff
@@ -0,0 +1,569 @@
+    x0 = x(E2V[0])
+    x1 = x(E2V[1])
+    y0 = y(E2V[0])
+    y1 = y(E2V[1])
+    z0 = z(E2V[0])
+    z1 = z(E2V[1])
+    # (xi, yi, zi) are normalized by construction
+
+    length = arc_length(x0, x1, y0, y1, z0, z1, radius)
+    return length
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED)
+def compute_edge_length(
+    vertex_lat: fa.VertexField[ta.wpfloat],
+    vertex_lon: fa.VertexField[ta.wpfloat],
+    radius: ta.wpfloat,
+    length: fa.EdgeField[ta.wpfloat],
+    horizontal_start: gtx.int32,
+    horizontal_end: gtx.int32,
+):
+    edge_length(
+        vertex_lat,
+        vertex_lon,
+        radius,
+        out=length,
+        domain={dims.EdgeDim: (horizontal_start, horizontal_end)},
+    )
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED)
+def compute_cell_center_arc_distance(
+    edge_neighbor_0_lat: fa.EdgeField[ta.wpfloat],
+    edge_neighbor_0_lon: fa.EdgeField[ta.wpfloat],
+    edge_neighbor_1_lat: fa.EdgeField[ta.wpfloat],
+    edge_neighbor_1_lon: fa.EdgeField[ta.wpfloat],
+    radius: ta.wpfloat,
+    dual_edge_length: fa.EdgeField[ta.wpfloat],
+    horizontal_start: gtx.int32,
+    horizontal_end: gtx.int32,
+):
+    cell_center_arc_distance(
+        lat_neighbor_0=edge_neighbor_0_lat,
+        lon_neighbor_0=edge_neighbor_0_lon,
+        lat_neighbor_1=edge_neighbor_1_lat,
+        lon_neighbor_1=edge_neighbor_1_lon,
+        radius=radius,
+        out=dual_edge_length,
+        domain={dims.EdgeDim: (horizontal_start, horizontal_end)},
+    )
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED)
+def compute_arc_distance_of_far_edges_in_diamond(
+    vertex_lat: fa.VertexField[ta.wpfloat],
+    vertex_lon: fa.VertexField[ta.wpfloat],
+    radius: ta.wpfloat,
+    far_vertex_distance: fa.EdgeField[ta.wpfloat],
+    horizontal_start: gtx.int32,
+    horizontal_end: gtx.int32,
+):
+    arc_distance_of_far_edges_in_diamond(
+        vertex_lat=vertex_lat,
+        vertex_lon=vertex_lon,
+        radius=radius,
+        out=far_vertex_distance,
+        domain={dims.EdgeDim: (horizontal_start, horizontal_end)},
+    )
+
+
+@gtx.field_operator
+def edge_area(
+    owner_mask: fa.EdgeField[bool],
+    primal_edge_length: fa.EdgeField[ta.wpfloat],
+    dual_edge_length: fa.EdgeField[ta.wpfloat],
+) -> fa.EdgeField[ta.wpfloat]:
+    """
+    Compute the area spanned by an edge and the its dual edge
+    Args:
+        owner_mask: owner mask for edges
+        primal_edge_length: length of edge in primal grid
+        dual_edge_length: length of edge in dual grid
+
+    Returns:
+        area
+
+    """
+    return where(owner_mask, primal_edge_length * dual_edge_length, 0.0)
+
+
+@gtx.program(grid_type=gtx.GridType.UNSTRUCTURED)
+def compute_edge_area(
+    owner_mask: fa.EdgeField[bool],
+    primal_edge_length: fa.EdgeField[ta.wpfloat],
+    dual_edge_length: fa.EdgeField[ta.wpfloat],
+    area: fa.EdgeField[ta.wpfloat],
+    horizontal_start: gtx.int32,
+    horizontal_end: gtx.int32,
+):
+    edge_area(
```

Comment:
in some field_operators call you have args and in some others keywargs. Can you put them all as keywargs?

Reply:
When calling field_ops in programs, I changed the input fields to positional args to make it constistent. The programs are only simple wrappers so the kwargs would not really help imho.